DESIGN.md

```diff
@@ -0,0 +1,108 @@
+# Vanagon
+- Note that this is an incomplete doc, documentation on all aspects of vanagon to come
+
+## Vanagon for the Windows platform
+
+### Overview
+Vanagon was adapted for the windows platform using WiX. At a very high level each component of the project gets built on a build system, then gets moved to a staging directory where we use the WiX toolset to build an MSI. This workflow is consistent with the rest of vanagon, however the WiX toolset is not so simple that there are no caveats, specifically:
+
+* Vanagon does not contain functionality to handle registry searching
+* Vanagon does not contain functionality to handle the MSI UI
+* If the project requires MSI custom actions (such as those to search the registry, set properties used in the MSI etc.) Vanagon has no functionality to handle those
+* For directories and files that require special attention (Such as a requirement for an app data directory to persist after uninstall) vanagon cannot handle them
+* Vanagon cannot handle the service files required by WiX for an MSI
+* Vanagon cannot handle the localization engine that MSIs use
```

Comment:
You also need to put in the very specific requirement that key Windows directories (e.g. Program files) must be specified using the install time dependancy localisation structure for windows. This is a key difference from the Unix style where we can specify in advance exactly where the files go, so we can actually build a replica of where the files go at build time, whereas with Windows we need to rely instead on install time relocatable root points.